optimum/habana/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion_xl.py

```diff
@@ -0,0 +1,843 @@
+# Copyright 2023 The HuggingFace Team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+#import inspect
+import time
+from dataclasses import dataclass
+from math import ceil
+from typing import Any, Callable, Dict, List, Optional, Tuple, Union
+
+import numpy as np
+import PIL
+import torch
+from diffusers.models import AutoencoderKL, UNet2DConditionModel
+from diffusers.pipelines.stable_diffusion_xl import StableDiffusionXLPipeline
+from diffusers.pipelines.stable_diffusion_xl.pipeline_stable_diffusion_xl import rescale_noise_cfg
+from diffusers.schedulers import KarrasDiffusionSchedulers
+from diffusers.utils import BaseOutput, deprecate
+from transformers import CLIPTextModel, CLIPTextModelWithProjection, CLIPTokenizer
+
+from optimum.utils import logging
+
+from ....transformers.gaudi_configuration import GaudiConfig
+from ....utils import speed_metrics
+from ..pipeline_utils import GaudiDiffusionPipeline
+
+
+@dataclass
+class GaudiStableDiffusionXLPipelineOutput(BaseOutput):
+    images: Union[List[PIL.Image.Image], np.ndarray]
+    throughput: float
+
+
+logger = logging.get_logger(__name__)  # pylint: disable=invalid-name
+
+EXAMPLE_DOC_STRING = """
+    Examples:
+        ```py
+        >>> import torch
+        >>> from optimum.habana.diffusers import GaudiStableDiffusionXLPipeline, GaudiDDIMScheduler
+
+        >>> kwargs = {
+        >>>     "scheduler": GaudiDDIMScheduler.from_pretrained(
+        >>>         "stabilityai/stable-diffusion-xl-base-1.0", subfolder="scheduler"
+        >>>     ),
+        >>>     "use_habana": True,
+        >>>     "use_hpu_graphs": True,
+        >>>     "gaudi_config": "Habana/stable-diffusion",
+        >>>     "torch_dtype": torch.bfloat16
+        >>> }
+
+        >>> pipe = GaudiStableDiffusionXLPipeline.from_pretrained(
+        >>>     "stabilityai/stable-diffusion-xl-base-1.0", **kwargs,
+        >>> )
+
+        >>> prompt = "a photo of an astronaut riding a horse on mars"
+        >>> image = pipe(prompt).images[0]
+        ```
+"""
```

Comment:
I don't think it is used anywhere so let's remove it

Reply:
Done